pandas/core/strings.py

```diff
@@ -2801,11 +2801,54 @@ def rindex(self, sub, start=0, end=None):
         return self._wrap_result(result)
 
     _shared_docs['len'] = ("""
-    Compute length of each string in the Series/Index.
+    Compute length of each element in the Series/Index. The element may be
+    a sequence (such as a string, tuple or list) or a collection
+    (such as a dictionary).
 
     Returns
     -------
-    lengths : Series/Index of integer values
+    Series or Index of integer values
```

Comment:
We usually use Python types, `Series or Index of int` would be more consistent with the rest of the docstrings.